jack.py

```diff
@@ -739,12 +739,21 @@ def set_process_callback(self, callback, userdata=None):
             is called.
 
         """
-        @self._callback("JackProcessCallback", error=STOP_CALLING)
-        def callback_wrapper(frames, _):
-            return callback(frames, userdata)
+        if isinstance(callback, _ffi.CData):
+            if _ffi.typeof(callback) != _ffi.typeof("JackProcessCallback"):
+                raise ValueError("Expected %s, got %s" % \
+                                 (_ffi.typeof("JackProcessCallback"),
+                                  _ffi.typeof(callback)))
```

Comment:
That's great, getting rid of the check with `ffi.typeof()` is even better!
CFFI indeed does a lot of checking, so we should do it only where it's really necessary.

Just out of curiosity: will `ffi.typeof()` really call the C function? I thought the information about the signature is available in the cdata object ...